File: Solutions/0278- First Bad Version/Solution.py
# The isBadVersion API is already defined for you.
# def isBadVersion(version: int) -> bool:

class Solution:
    def firstBadVersion(self, n: int) -> int:
        left = 1
        right = n

        while left < right:
            middle = (left + right) >> 1
            if not isBadVersion(middle):
                left = middle + 1
            else:
                right = middle

        return left

File: Solutions/0278- First Bad Version/Solution.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

      public class Solution extends VersionControl {
        public int firstBadVersion(int n) {
            int left = 1;
            int right = n;
    
            while (left < right) {
                int middle = (left + right) >>> 1;
            
                if (!isBadVersion(middle))
                    left = middle + 1;
                else {
                    right = middle;
                }
                    
            }
    
            return left;
        }
    }